src/Model/Resolver/GenerateCustomerToken.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/module-customer-graph-ql
+ * @link https://github.com/scandipwa/module-customer-graph-ql
+ */
+declare(strict_types=1);
+
+namespace ScandiPWA\CustomerGraphQl\Model\Resolver;
+
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthenticationException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Integration\Api\CustomerTokenServiceInterface;
+use Magento\Framework\Event\ManagerInterface as EventManager;
+
+/**
+ * Customers Token resolver, used for GraphQL request processing.
+ */
+class GenerateCustomerToken implements ResolverInterface
+{
+    /**
+     * @var CustomerTokenServiceInterface
+     */
+    private $customerTokenService;
+
+    /**
+     * @var EventManager
+     */
+    private $eventManager;
+
+    /**
+     * @param CustomerTokenServiceInterface $customerTokenService
+     */
+    public function __construct(
+        CustomerTokenServiceInterface $customerTokenService,
+        EventManager $eventManager
+    ) {
+        $this->customerTokenService = $customerTokenService;
+        $this->eventManager = $eventManager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($args['email']) || empty($args['email'])) {
+            throw new GraphQlInputException(__('Specify the "email" value.'));
+        }
+
+        if (!isset($args['password']) || empty($args['password'])) {
+            throw new GraphQlInputException(__('Specify the "password" value.'));
+        }
+
+        if (!isset($args['guest_quote_token']) || empty($args['guest_quote_token'])) {
+            throw new GraphQlInputException(__('"guest_quote_token" value is not specified.'));
+        }
+
+        try {
+            $customerToken = $this->customerTokenService->createCustomerAccessToken($args['email'], $args['password']);
+
+            $this->eventManager->dispatch('generate_customer_token_after', [
+                'guest_token' => $args['guest_quote_token'],
```

Comment:
confusing naming. it goes from guest_quote_id -> guest_quote_token -> guest_token. I would stick with just guest_quote_id everywhere